.github/workflows/release.yml

```diff
@@ -4,35 +4,56 @@ on:
   push:
     branches:
       - main
-  workflow_run:
-    workflows: ['Lints', 'Coverage']
-    types:
-      - completed
+      - development
 
 concurrency: ${{ github.workflow }}-${{ github.ref }}
 
 jobs:
   release:
     name: Release
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: write
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
       - name: Setup Node.js
-        uses: actions/setup-node@v3
+        uses: actions/setup-node@v4
         with:
-          node-version: 16
-          cache: 'yarn'
-          cache-dependency-path: '**/yarn.lock'
-
-      - uses: actions/setup-node@v1
-        with:
-          node-version: 16
+          node-version: '20.x'
           registry-url: 'https://registry.npmjs.org'
-      - name: Config git user
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Run build
+        run: yarn lerna run build
+
+      - name: Run test
+        run: yarn test
+
+      - name: Configure git user
         run: |
           git config --global user.name "${{ github.actor }}"
           git config --global user.email "${{ github.actor }}@users.noreply.github.com"
 
-      - run: yarn install --frozen-lockfile
+      - name: Bump version and create release
+        run: |
+          if [ "${{ github.ref_name }}" == "development" ]; then
+            yarn lerna version --yes --conventional-commits --conventional-prerelease --exact --create-release github --preid beta --loglevel silly
+          else
+            yarn lerna version --yes --conventional-commits --conventional-graduate --exact --create-release github --loglevel silly
+          fi
+        env:
+          GH_TOKEN: ${{ github.token }}
+
+      - name: Publish to NPM
+        run: |
+          if [ "${{ github.ref_name }}" == "development" ]; then
+            yarn lerna publish from-package --yes --no-private --loglevel silly --preid beta --dist-tag beta
+          else
+            yarn lerna publish from-package --yes --no-private --loglevel silly
```

Comment:
Also here

Reply:
This is handled by [lerna.json](https://github.com/ngraveio/ur-registry/blob/ec23ff33d1bf96c2f0479d620840333b9aaee7b8/lerna.json#L9) and fails if it runs on a branch other than main and release